src/promptflow/promptflow/core/_flow.py

```diff
@@ -0,0 +1,322 @@
+
+        result = self.invoke(inputs=kwargs)
+        return result.output
+
+    def invoke(self, inputs: dict) -> "LineResult":
+        """Invoke a flow and get a LineResult object."""
+        from promptflow.core._flow_context_resolver import FlowContextResolver
+
+        invoker = FlowContextResolver.resolve(flow=self)
+        result = invoker._invoke(
+            data=inputs,
+        )
+        return result
+
+
+class EagerFlow(FlowBase):
+    """This class is used to represent an eager flow."""
+
+    def __init__(
+        self,
+        path: Union[str, PathLike],
+        code: Union[str, PathLike],
+        entry: str,
+        data: dict,
+        **kwargs,
+    ):
+        # flow.dag.yaml file path or entry.py file path
+        path = Path(path)
+        # flow.dag.yaml file's folder or entry.py's folder
+        code = Path(code)
+        # entry function name
+        self.entry = entry
+        # entry file name
+        self.entry_file = self._resolve_entry_file(entry=entry, working_dir=code)
+        # TODO(2910062): support eager flow execution cache
+        super().__init__(data=data, path=path, code=code, content_hash=None, **kwargs)
+
+    @classmethod
+    def _load(cls, path: Path, data: dict, **kwargs):
+        entry = data.get("entry")
+        code = path.parent
+
+        if entry is None:
+            raise UserErrorException(f"Entry function is not specified for flow {path}")
+        return cls(path=path, code=code, entry=entry, data=data, **kwargs)
+
+    @classmethod
+    def _resolve_entry_file(cls, entry: str, working_dir: Path) -> Optional[str]:
+        """Resolve entry file from entry.
+        If entry is a local file, e.g. my.local.file:entry_function, return the local file: my/local/file.py
+            and executor will import it from local file.
+        Else, assume the entry is from a package e.g. external.module:entry, return None
+            and executor will try import it from package.
+        """
+        try:
+            entry_file = f'{entry.split(":")[0].replace(".", "/")}.py'
+        except Exception as e:
+            raise UserErrorException(f"Entry function {entry} is not valid: {e}")
+        entry_file = working_dir / entry_file
+        if entry_file.exists():
+            return entry_file.resolve().absolute().as_posix()
+        # when entry file not found in working directory, return None since it can come from package
+        return None
+
+    def _init_executable(self, **kwargs):
+        from promptflow.contracts.flow import EagerFlow as ExecutableEagerFlow
+
+        # TODO(2991934): support environment variables here
+        meta_dict = generate_flow_meta(
+            flow_directory=self.code,
+            source_path=self.entry_file,
+            entry=self.entry,
+            dump=False,
+        )
+        return ExecutableEagerFlow.deserialize(meta_dict)
+
+
+class AsyncFlow(Flow):
```

Comment:
should have a load function?

Reply:
same with _load, it just override the Flow.load()
Flow.load could separate the Flows
EagerFlow.load could not.